packages/web-components/src/components/quote/quote.ts

```diff
@@ -0,0 +1,183 @@
+    switch (this.markType) {
+      case QUOTE_TYPES.SINGLE_CURVED:
+        return html`
+          <span class="${prefix}--quote__mark">‘</span>
+          <blockquote class="${prefix}--quote__copy"><slot name="copy"></slot>’</blockquote>
+        `;
+      case QUOTE_TYPES.DOUBLE_ANGLE:
+        return html`
+          <span class="${prefix}--quote__mark">«</span>
+          <blockquote class="${prefix}--quote__copy"><slot name="copy"></slot>»</blockquote>
+        `;
+      case QUOTE_TYPES.SINGLE_ANGLE:
+        return html`
+          <span class="${prefix}--quote__mark">‹</span>
+          <blockquote class="${prefix}--quote__copy"><slot name="copy"></slot>›</blockquote>
+        `;
+      case QUOTE_TYPES.LOW_HIGH_REVERSED_DOUBLE_CURVED:
+        return html`
+          <span class="${prefix}--quote__mark">„</span>
+          <blockquote class="${prefix}--quote__copy"><slot name="copy"></slot>“</blockquote>
+        `;
+      default:
+        return html`
+          <span class="${prefix}--quote__mark">“</span>
+          <blockquote class="${prefix}--quote__copy"><slot name="copy"></slot>”</blockquote>
+        `;
+    }
+  }
+
+  _renderSource() {
+    const {
+      _hasSourceHeading: hasSourceHeading,
+      _hasSourceCopy: hasSourceCopy,
+      _hasSourceBottomCopy: hasSourceBottomCopy,
+      _handleSlotChange: handleSlotChange,
+    } = this;
+    return html`
+      <div ?hidden="${!hasSourceHeading || !hasSourceCopy}" class="${prefix}--quote__source">
+        <p class="${prefix}--quote__source-heading">
+          <slot @slotchange="${handleSlotChange}" name="sourceHeading" />
+        </p>
+        <p class="${prefix}--quote__source-body">
+          <slot @slotchange="${handleSlotChange}" name="sourceCopy" />
+        </p>
+        <p ?hidden="${!hasSourceBottomCopy}" class="${prefix}--quote__source-optional-copy">
+          <slot @slotchange="${handleSlotChange}" name="sourceBottomCopy" />
+        </p>
+      </div>
+    `;
+  }
+
+  _renderCTA() {
+    const { _hasCTA: hasCTA, _handleSlotChange: handleSlotChange } = this;
+    return html`
+      <div ?hidden="${!hasCTA}" class="${prefix}--quote__footer">
+        <dds-hr></dds-hr>
+        <slot name="cta" @slotchange="${handleSlotChange}"></slot>
+      </div>
+    `;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  render() {
+    return html`
+      <div class="${prefix}--quote__container">
+        <div class="${prefix}--quote__wrapper">
+          ${this._renderQuote()} ${this._renderSource()} ${this._renderCTA()}
+        </div>
+      </div>
+    `;
+  }
+
+  /**
+   * Handles `slotchange` event.
+   */
+  protected _handleSlotChange({ target }: Event) {
+    const { name } = target as HTMLSlotElement;
+    const hasContent = (target as HTMLSlotElement)
+      .assignedNodes()
+      .some(node => node.nodeType !== Node.TEXT_NODE || node!.textContent!.trim());
+    this[slotExistencePropertyNames[name]] = hasContent;
+    this.requestUpdate();
+  }
```

Comment:
Please move protected properties/methods up above, to align to the rest of classes in the codebase.